src/Transports/TransportZmq.cpp

```diff
@@ -51,8 +51,15 @@ void TransportZmq::connect() {
         switch (type) {
             case REQ: {
                 socket = std::shared_ptr<zmq::socket_t>(new zmq::socket_t(*(context.get()), ZMQ_REQ));
-                socket->setsockopt(ZMQ_REQ_CORRELATE, 1);
-                socket->setsockopt(ZMQ_REQ_RELAXED, 1);
+
+                #if ZMQ_VERSION < ZMQ_MAKE_VERSION(4, 7, 0)
```

Comment:
Ubuntu 24 installs ZMQ version 4.3.5, but still gives the warning that `setsockopt` is deprecated 
```
robot_remote_control/src/Transports/TransportZmq.cpp:56:39: warning: ‘void zmq::detail::socket_base::setsockopt(int, const T&) [with T = int]’ is deprecated: from 4.7.0, use `set` taking option from zmq::sockopt [-Wdeprecated-declarations]
   56 |                     socket->setsockopt(ZMQ_REQ_CORRELATE, 1);
      |                     ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
```

Maybe checking `#ifndef ZMQ_CPP11` (as you did before) is the intended way and the deprecation warning actually means that with version 4.7.0 these functions will be removed? (I don't know).

Reply:
Checking against ZMQ_CPP11 did not work, it is also set in 20.04. But the version should be checked against the cpp header version not the zmq lib version: 

`CPPZMQ_VERSION< ZMQ_MAKE_VERSION(4, 7, 0)`

instead of

`ZMQ_VERSION< ZMQ_MAKE_VERSION(4, 7, 0)`

Reply:
Looks fine now on Ubuntu 24, but apparently fails again on Ubuntu 20?

Reply:
20.04 comes with exactly 4.7.0, which is "old style", so the check has to be "<=" instead of "<", new build is running

Reply:
Slightly confusing deprecation warning "from 4.7.0, use `set`" then. But I guess we can do it like this then.